linera-chain/src/manager.rs

```diff
@@ -433,14 +446,16 @@ impl ChainManager {
             } else {
                 HashedCertificateValue::new_validated(executed_block)
             };
-            self.pending = Some(Vote::new(value, round, key_pair));
+            self.pending_vote = Some(Vote::new(value, round, key_pair));
         }
     }
 
     /// Signs a vote to confirm the validated block.
     pub fn create_final_vote(
         &mut self,
         validated: ValidatedBlockCertificate,
+        published_blobs: Vec<Blob>,
+        used_blobs: Vec<Blob>,
```

Comment:
both published and used blobs should be part of the block which is already part of the `ValidatedBlockCertificate` - we shouldn't be passing them in addition to `validated` as it makes it easy to make a mistake.

Reply:
Only their hashes are part of the certificate.